server/routes/contacts/manage/contact-search/contactSearchSchema.ts

```diff
@@ -0,0 +1,76 @@
+import z from 'zod'
+import { createSchema } from '../../../../middleware/validationMiddleware'
+
+const LAST_NAME_REQUIRED_MESSAGE = "Enter the contact's last name"
+const LAST_NAME_INVALID = "Contact's last name must not contain special characters"
+const FIRST_NAME_INVALID = "Contact's first name must not contain special characters"
+const MIDDLE_NAME_INVALID = "Contact's middle name must not contain special characters"
+const DAY_TYPE_MESSAGE = 'Enter a valid day of the month (1-31)'
+const MONTH_TYPE_MESSAGE = 'Enter a valid month (1-12)'
+const YEAR_TYPE_MESSAGE = 'Enter a valid year. Must be at least 1900'
+const DOB_IN_FUTURE_MESSAGE = 'The date of birth must not be in the future'
+
+const NAME_REGEX = /^[a-zA-Z\s,.'-]*$/
+
+export const contactSearchSchema = () => async () => {
```

Comment:
I wonder if we should loosen this schema a bit? It's nice to have it very strict on create journey as it prevents bad data going in, however, we'll be inheriting already bad data from NOMIS and this would prevent the user for searching for it... what do you think @timharrison-moj ?

If we do want to keep strict rules we should maybe build some shared definitions so we don't need to copy paste the tests or modify multiple places if a change is required, e.g., if we want to allow additional characters in a name, it's now in two places.